packages/browser/src/login/oidc/oidcHandlers/AuthorizationCodeWithPkceOidcHandler.ts

```diff
@@ -59,36 +61,57 @@ export default class AuthorizationCodeWithPkceOidcHandler
-      response_type: "code id_token",
-      redirect_uri: oidcLoginOptions.redirectUrl.toString(),
-      // TODO: the 'webid' scope does not appear in the specification
-      // A question regarding its use has been filed https://github.com/solid/specification
-      scope: "openid webid offline_access",
+    const oidcOptions = {
+      authority: oidcLoginOptions.issuer?.toString(),
       client_id: oidcLoginOptions.client.clientId,
-      code_challenge_method: "S256",
-      code_challenge: await this.joseUtility.generateCodeChallenge(
-        codeVerifier
-      ),
-      state: oidcLoginOptions.sessionId,
+      client_secret: oidcLoginOptions.client.clientSecret,
+      redirect_uri: oidcLoginOptions.redirectUrl?.toString(),
+      post_logout_redirect_uri: oidcLoginOptions.redirectUrl?.toString(),
+      response_type: "code",
+      scope: "openid webid",
+      filterProtocolClaims: true,
+      loadUserInfo: true,
+      code_verifier: true,
     };
     /* eslint-enable @typescript-eslint/camelcase */
-    requestUrl.set("query", query);
-    // TODO: This is inefficent, there should be a bulk
-    await this.storageUtility.setForUser(oidcLoginOptions.sessionId, {
-      codeVerifier,
-      issuer: oidcLoginOptions.issuer.toString(),
-      redirectUri: oidcLoginOptions.redirectUrl.toString(),
-    });
 
-    this.redirector.redirect(requestUrl.toString(), {
-      handleRedirect: oidcLoginOptions.handleRedirect,
-    });
+    const oidcClientLibrary = new OidcClient(oidcOptions);
+
+    oidcClientLibrary
+      .createSigninRequest()
+      .then(async function (req: SigninRequest) {
+        // We use the OAuth 'state' value (which should be crypto-random) as
+        // the key in our storage to store our actual SessionID. We do this 'cos
+        // we'll need to lookup our session information again when the browser
+        // is redirected back to us (i.e. the OAuth client application) from the
+        // Authorization Server.
+        // We don't want to use our session ID as the OAuth 'state' value, as
+        // that session ID can be any developer-specified value, and therefore
+        // may not be appropriate (since the OAuth 'state' value should really
+        // be an unguessable crypto-random value).
+        await storage.setForUser(req.state._id, {
+          sessionId: oidcLoginOptions.sessionId,
+          // codeVerifier: req.state._code_verifier,
+          // issuer: oidcLoginOptions.issuer.toString(),
+          // redirectUri: oidcLoginOptions.redirectUrl.toString(),
+        });
+
+        await storage.setForUser(oidcLoginOptions.sessionId, {
+          codeVerifier: req.state._code_verifier,
+          issuer: oidcLoginOptions.issuer.toString(),
+          redirectUri: oidcLoginOptions.redirectUrl.toString(),
+        });
+
+        redirector.redirect(req.url.toString(), {
+          handleRedirect: oidcLoginOptions.handleRedirect,
+        });
+      })
+      .catch((err: any) => {
```

Comment:
There's an ESLint warning here:

```suggestion
      .catch((err: unknown) => {
```